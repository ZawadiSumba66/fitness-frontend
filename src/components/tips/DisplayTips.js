import { useEffect } from 'react';
import { connect, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { Carousel, Spinner } from 'react-bootstrap';
import { fetchTips } from '../../actions/tip_action';
import DisplayTip from './DisplayTip';
import DashboardLeft from '../user/DashboardLeft';
import Flash from '../user/Flash';

const DisplayTips = ({ tips }) => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchTips());
  }, []);
  if (!tips) {
    return (
      <div className="d-flex justify-content-center align-items-center spinner__wrapper">
        <Spinner animation="border" variant="primary" role="status">
          <span className="sr-only">Loading...</span>
        </Spinner>
      </div>
    );
  }
  return (
    <div>
      <DashboardLeft />
      <div className="user__info p-3">
        <div className="rounded p-3 mt-4">
          <Flash />
        </div>
      </div>
      <div className="d-flex flex-column mt-10 mb-5">
        <div>
          <Carousel data-ride="carousel">
            {tips.map((tip) => (
              <Carousel.Item key={tip.id}>
                <DisplayTip
                  key={tip.id}
                  id={tip.id}
                  title={tip.title}
                  description={tip.description}
                />
              </Carousel.Item>
            ))}
          </Carousel>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({ tips: state.tipsReducer.tips });

DisplayTips.propTypes = {
  tips: PropTypes.instanceOf(Array).isRequired,
};

export default connect(mapStateToProps, null)(DisplayTips);
