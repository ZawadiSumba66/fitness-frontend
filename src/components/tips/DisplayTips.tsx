import { useEffect } from 'react';
import { connect, useDispatch } from 'react-redux';
import { Carousel, Spinner } from 'react-bootstrap';
import { fetchTips, UserTips } from '../../actions/tip_action';
import DisplayTip from './DisplayTip';
import DashboardLeft from '../user/DashboardLeft';
import Flash from '../user/Flash';

type TipsDisplay = {
  tips: UserTips[]
};

const DisplayTips: React.FunctionComponent<any> = ({ tips }: TipsDisplay) => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchTips());
  }, []);
  if (!tips) {
    return (
      <div className="d-flex justify-content-center align-items-center spinner__wrapper">
        <Spinner animation="border" variant="primary" role="status">
          <span className="sr-only">Loading...</span>
        </Spinner>
      </div>
    );
  }
  return (
    <div>
      <DashboardLeft />
      <div className="user__info p-3">
        <div className="rounded p-3 mt-4">
          <Flash />
        </div>
      </div>
      <div className="d-flex flex-column mt-10 mb-5">
        <div>
          <Carousel data-ride="carousel">
            {tips.map((tip: any) => (
              <Carousel.Item key={tip.id}>
                <DisplayTip
                  key={tip.id}
                  id={tip.id}
                  title={tip.title}
                  description={tip.description}
                />
              </Carousel.Item>
            ))}
          </Carousel>
        </div>
      </div>
    </div>
  );
};

type StateTips = {
  tipsReducer: {
    tips: string[]
  }
};

const mapStateToProps = (state: StateTips) => ({ tips: state.tipsReducer.tips });

export default connect(mapStateToProps, null)(DisplayTips);
