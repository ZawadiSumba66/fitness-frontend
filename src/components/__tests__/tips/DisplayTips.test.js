import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../../store';
import DisplayTips from '../../tips/DisplayTips';

it('matches the Display Tips snapshot', () => {
  const tree = renderer.create(<Provider store={store}><DisplayTips /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
