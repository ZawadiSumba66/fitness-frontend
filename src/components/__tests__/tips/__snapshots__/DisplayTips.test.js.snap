// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the Display Tips snapshot 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center spinner__wrapper"
  >
    <div
      className="spinner-border text-primary"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
  <div
    className="user__info p-3"
  >
    <div
      className="rounded p-3 mt-4"
    >
      <div />
    </div>
  </div>
  <div
    className="d-flex flex-column mb-5"
  >
    <div>
      <div
        className="carousel slide"
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <ol
          className="carousel-indicators"
        />
        <div
          className="carousel-inner"
        />
        <a
          className="carousel-control-prev"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            aria-hidden="true"
            className="carousel-control-prev-icon"
          />
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            aria-hidden="true"
            className="carousel-control-next-icon"
          />
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
